### ISSUE TITLE
Error when attempting to make a JAR file executable and then executing it

### ISSUE THREAD

**Issue by etxaleku** [adoptium/temurin-build#3670]
I am trying to execute a jar file produced from IntelliJ.  

The file executes correctly using the traditional command line -

```java -jar myprogram.jar```

However, after making myprogram.jar executable and then attempting to execute, it generates an error -

```
$ chmod +x myprogram.jar
$ myprogram.jar
Error: Could not find or load main class .path.to.myfile.myprogram.jar
Caused by: java.lang.ClassNotFoundException: /path/to/myfile/myprogram/jar
```

Operating system: Ubuntu 22.04.3 LTS
(looks like there are many openjdk packages installed, probably through dependencies)
The Java I am using is installed through SDKMAN and is 
```
$ java --version
openjdk 21.0.1 2023-10-17 LTS
OpenJDK Runtime Environment Temurin-21.0.1+12 (build 21.0.1+12-LTS)
OpenJDK 64-Bit Server VM Temurin-21.0.1+12 (build 21.0.1+12-LTS, mixed mode, sharing)
$ echo $JAVA_HOME
/usr/local/sdkman/candidates/java/current
$ sdk home java current
/usr/local/sdkman/candidates/java/current
$ type java
java is /usr/local/sdkman/candidates/java/current/bin/java
```

In order to get Linux to execute a jar file (directly) you first need to install binfmt-support.  It basically sets up the Linux magic number-to-interpreter mappings for various (configurable) interpreters.

It looks like some of the Ubuntu openjdk packages are integrated with binfmt-support as the default configuration looked like -

```
# update-binfmts --display jar
jar (enabled):
     package = openjdk-11
        type = magic
      offset = 0
       magic = PK\x03\x04
        mask =
 interpreter = /usr/bin/jexec
    detector =
```

So I updated the configuration to look like -

```
# update-binfmts --display jar
jar (enabled):
     package = temurin-21.0.1
        type = magic
      offset = 0
       magic = PK\x03\x04
        mask =
 interpreter = /usr/local/sdkman/candidates/java/current/bin/java
    detector =
```

I basically made up the package name as I don't use the Linux package manager to manage my JDK versions but use SDKMAN instead.

But I checked the jar file and the first four bytes are PK\x03\x04 ...

It seems to me as though my Temurin java is having trouble with the MANIFEST.MF - which has the Main-Class specified correctly -

```
$ jar xf myprogram.jar META-INF/MANIFEST.MF ; cat META-INF/MANIFEST.MF
Manifest-Version: 1.0
Main-Class: com.myorg.myprogram
```

Not sure this qualifies as an issue but I did notice that older openjdk packages provided a binfmt config file that looks like it gets applied during installation and thought maybe Temurin would want to do the same.

```
$ cat /etc/alternatives/jexec-binfmt
package openjdk-11
interpreter /usr/bin/jexec
magic PK\x03\x04
```

**Comment by sxa** [comment]
This should probably be in the support repository rather than build since this isn't something that would come out of the build as such (although it could be incorporated into the deb/rpm packages), but I can move it if necessary.

It looks like this is an Ubuntu specific configuration, so my gut feel is that we probably wouldn't replicate this as-is. Regarding your "updated" binfmt configuration is there a reason you pointed the `interpreter` line at `java` instead of `jexec` which is what the Ubuntu one seemed to be set to?

**Comment by etxaleku** [comment]
Ah, the support repository makes sense.  I just clicked on the "Issues" button, not sure how to get to the support repository.

The only reason I pointed the interpreter to java vs jexec was because I didn't see a jexec in my Temurin java installation.  I only looked in /usr/local/sdkman/candidates/java/current/bin/ :( and your feedback seemed confident there would be one so a quick find located one in the lib directory (sorry, should have done that earlier) -

/usr/local/sdkman/candidates/java/current/lib/jexec

After updating the binfmts config file to point to the newly located jexec everything is now working!

Thanks for the education and getting me back on track!!  :)
